desktop/plugins/network/index.tsx

```diff
@@ -387,6 +404,40 @@ export default class extends FlipperPlugin<State, any, PersistedState> {
           }),
         );
       },
+      copyHighlightedCalls(
+        highlightedRows: Set<string> | null | undefined,
+        requests: {[id: string]: Request},
+        responses: {[id: string]: Response},
+      ) {
+        setState(
+          produce((draftState: State) => {
+            // iterate through highlighted rows
+            highlightedRows?.forEach((row) => {
+              const response = responses[row];
+              // convert headers
+              const headers: {[id: string]: Header} = {};
+              response.headers.forEach((e) => {
+                headers[e.key] = e;
+              });
+
+              // convert data
+              const responseData =
+                response && response.data ? decodeBody(response) : null;
+
+              const nextRouteId = draftState.nextRouteId;
+              draftState.routes[nextRouteId.toString()] = {
+                requestUrl: requests[row].url,
+                requestMethod: requests[row].method,
+                responseData: responseData as string,
+                responseHeaders: headers,
+                responseStatus: responses[row].status.toString(),
+              };
+              draftState.nextRouteId = nextRouteId + 1;
+              informClientMockChange(draftState.routes);
```

Comment:
This is an awesome feature! Thanks for building it :)

Regarding `informClientMockChange`: This should best be done after the produce, as otherwise the draft is leaked from the produce, which will throw exceptions if any async reference is made the to the routes. So this would become:

```
setState(state => {
   const nextState = produce(state, /* same fn */)
   informClientMockChange(nextState.routes)
   return nextState
})
```

Note that using a side effect in setState is an anti pattern in itself, so instead of the above, the idiomatic thing to do is `useEffect(() => { informClientMockChange(nextState.routes) }, [nextState.routes])`, but that is a higher impact change as multiple code paths lead there, so I can live with the first proposed solution as well

Reply:
@mweststrate Thanks for the feedback.  I'm an Android developer so struggling a bit with the JavaScript.

I made the changes you suggested (I think!) but having some issues with it.  Here's the code for the function "copyHighlightedCalls" along with the error message.  I hoping you can give me some direction on it.  Thanks

![image](https://user-images.githubusercontent.com/69264583/92000349-0acb4580-ed03-11ea-8f48-54ef7d0f45f4.png)

Reply:
should be `produce(state, (state: State) => /* etc */`

Reply:
@mweststrate You're a guru!  Made the change and tested it.  Just pushed so it should show up shortly.